src/containerapp-preview/azext_containerapp_preview/_utils.py

```diff
@@ -0,0 +1,90 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=line-too-long
+
+from azure.cli.core.azclierror import NoTTYError
+from azure.cli.core.extension.dynamic_install import _get_extension_use_dynamic_install_config
+from knack.prompting import prompt_y_n, NoTTYException
+from knack.util import CLIError
+from knack.log import get_logger
+
+logger = get_logger(__name__)
+
+MIN_GA_VERSION = '0.3.35'
+
+
+def _get_or_add_extension(cmd, extension_name):
+    from azure.cli.core.extension import (
+        ExtensionNotInstalledException, get_extension)
+    from pkg_resources import parse_version
```

Comment:
Please use `packaging.version` instead of `pkg_resources`. Context: https://github.com/Azure/azure-cli/pull/17667/files
You can follow this PR https://github.com/Azure/azure-cli/pull/26797 as an example